guidance/models/_model_state.py

```diff
@@ -0,0 +1,129 @@
+from __future__ import annotations
+from dataclasses import dataclass
+from typing import Optional, overload
+import base64
+import html
+
+
+@dataclass(frozen=True, slots=True)
+class Object:
+    def _html(self) -> str:
+        raise NotImplementedError
+
+    def __str__(self) -> str:
+        raise NotImplementedError
```

Comment:
Currently undecided as to whether these should be methods or if these should just be data containers with external "Handler" classes to produce HTML, strings for prompts, pretty ANSI codes for terminal outputs, etc...

Reply:
I'd definitely separate model state from how it gets displayed.